core/src/admin_rpc_post_init.rs

```diff
@@ -8,19 +8,79 @@ use {
     solana_quic_definitions::NotifyKeyUpdate,
     solana_runtime::bank_forks::BankForks,
     std::{
-        collections::HashSet,
+        collections::{HashMap, HashSet},
         net::UdpSocket,
         sync::{Arc, RwLock},
     },
 };
 
+/// Key updaters:
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub enum KeyUpdaterType {
+    /// TPU key updater
+    Tpu,
+    /// TPU forwards key updater
+    TpuForwards,
+    /// TPU vote key updater
+    TpuVote,
+    /// Forward key updater
+    Forward,
+    /// For the RPC service
+    RpcService,
+    /// For the connection cache
+    ConnectionCache,
+}
+
+/// Responsible for managing the updaters for identity key change
+#[derive(Default)]
+pub struct KeyUpdaters {
+    updaters: HashMap<KeyUpdaterType, Arc<dyn NotifyKeyUpdate + Sync + Send>>,
```

Comment:
I wonder if we really need to have `HashMap`. Although it is it is not performance-sensitive code, I think that using plain array would work. Something like (haven't checked):

```rust
#[repr(usize)]
pub enum KeyUpdaterType {
    Tpu = 0,
    TpuForwards,
    TpuVote,
    Forward,
    RpcService,
    ConnectionCache,
}

pub struct KeyUpdaters {
    updaters: [Option<Arc<dyn NotifyKeyUpdate + Send + Sync>>; NUM_UPDATERS],
}
```

Reply:
I think this would work. But in my opinion it is less clean -- we would have to check none, and translating between the integer and enum in the code. This is very low frequency event. Using a map is simpler to reason about.

Reply:
Yeah, not a real problem since this code is used almost never. Maybe someone should create a container with nice interface for fixed size HashMaps.

Reply:
https://docs.rs/fixed-map/latest/fixed_map/ can be considered for high frequency  use cases